File: BrainBoxAPI/BrainBoxAPI/Models/Document.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace BrainBoxAPI.Models
{
    public class Document
    {
        [Key]
        public int DocId { get; set; }

        public string Title { get; set; }

        public string Content { get; set; }

        [ForeignKey("User")]
        public int AuthorId { get; set; }

        public bool IsPublic { get; set; }

        public int Views { get; set; }

        public long CreatedAt { get; set; }

        public User? User { get; set; }
    }
}

File: PRM392_BrainBox_v1.0/app/src/main/java/com/example/prm392_v1/data/entity/Document.java
package com.example.prm392_v1.data.entity;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "documents")
public class Document {
    @PrimaryKey(autoGenerate = true)
    public int docId;

    public String title;

    public String content;

    public int authorId;

    public boolean isPublic = true;

    public int views = 0;

    public long createdAt = System.currentTimeMillis();
}
